In [1]:
import nsfg1
import numpy as np
import pandas as pd
from importlib import reload
reload(nsfg1)

df_preg = nsfg1.read_fem_preg()
df_preg

caseid  pregordr  howpreg_n  howpreg_p  moscurrp  nowprgdk  pregend1  \
0      60418         1        NaN        NaN       NaN       NaN       5.0   
1      60418         2        NaN        NaN       NaN       NaN       5.0   
2      60418         3        NaN        NaN       NaN       NaN       5.0   
3      60419         1       33.0        1.0       8.0       NaN       NaN   
4      60420         1        NaN        NaN       NaN       NaN       6.0   
...      ...       ...        ...        ...       ...       ...       ...   
9353   70615         3        NaN        NaN       NaN       NaN       6.0   
9354   70616         1        NaN        NaN       NaN       NaN       5.0   
9355   70616         2        NaN        NaN       NaN       NaN       5.0   
9356   70619         1        NaN        NaN       NaN       NaN       6.0   
9357   70619         2        NaN        NaN       NaN       NaN       5.0   

      pregend2  howenddk  nbrnaliv  ...  cmintvw  cmlstyr  cmjan3yr  cmjan4yr  \
0          NaN       NaN       1.0  ...      NaN     1369      1345      1333   
1          NaN       NaN       1.0  ...      NaN     1369      1345      1333   
2          NaN       NaN       1.0  ...      NaN     1369      1345      1333   
3          NaN       NaN       NaN  ...      NaN     1376      1345      1333   
4          NaN       NaN       1.0  ...      NaN     1376      1345      1333   
...        ...       ...       ...  ...      ...      ...       ...       ...   
9353       NaN       NaN       1.0  ...      NaN     1353      1321      1309   
9354       NaN       NaN       1.0  ...      NaN     1375      1345      1333   
9355       NaN       NaN       1.0  ...      NaN     1375      1345      1333   
9356       NaN       NaN       1.0  ...      NaN     1370      1345      1333   
9357       NaN       NaN       1.0  ...      NaN     1370      1345      1333   

      cmjan5yr  quarter  phase  intvwyear  totalwgt_lb1  totalwgt_kg  
0         1321       14      1        201        5.2500     2.381358  
1         1321       14      1        201        4.7500     2.154562  
2         1321       14      1        201        5.2500     2.381358  
3         1321       16      1        201           NaN          NaN  
4         1321       16      1        201        8.8125     3.997279  
...        ...      ...    ...        ...           ...          ...  
9353      1297        9      1        201        4.6875     2.126212  
9354      1321       16      1        201        8.4375     3.827183  
9355      1321       16      1        201        7.3750     3.345241  
9356      1321       14      1        201        8.1875     3.713785  
9357      1321       14      1        201        8.3125     3.770484  

[9358 rows x 280 columns]

In [2]:
df_preg['birthwgt_lb1']

0       5.0
1       4.0
2       5.0
3       NaN
4       8.0
       ... 
9353    4.0
9354    8.0
9355    7.0
9356    8.0
9357    8.0
Name: birthwgt_lb1, Length: 9358, dtype: float64

In [3]:
df_preg.columns

Index(['caseid', 'pregordr', 'howpreg_n', 'howpreg_p', 'moscurrp', 'nowprgdk',
       'pregend1', 'pregend2', 'howenddk', 'nbrnaliv',
       ...
       'cmintvw', 'cmlstyr', 'cmjan3yr', 'cmjan4yr', 'cmjan5yr', 'quarter',
       'phase', 'intvwyear', 'totalwgt_lb1', 'totalwgt_kg'],
      dtype='object', length=280)

In [4]:
df_preg.columns[1]

'pregordr'

In [5]:
pregordr = df_preg['pregordr']
type(pregordr)

pandas.core.series.Series

In [6]:
pregordr[0]

1

In [7]:
pregordr[2:5]

2    3
3    1
4    1
Name: pregordr, dtype: int64

In [8]:
pregordr = df_preg.pregordr

In [9]:
np.nan / 100.0

nan

In [10]:
df_preg.prgoutcome.value_counts().sort_index()

prgoutcome
1.0    6485
2.0    2621
3.0     249
Name: count, dtype: int64

In [11]:
df_preg.birthwgt_lb1.value_counts(sort=False)

birthwgt_lb1
5.0      586
4.0      196
8.0     1168
6.0     1666
7.0     2146
3.0       67
10.0      82
1.0       34
9.0      363
11.0      17
12.0       7
2.0       47
14.0       2
0.0        6
13.0       2
17.0       1
Name: count, dtype: int64

indices is the list of indices for pregnancies corresponding to respondent 9353

In [12]:
print(dir(nsfg1))

['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'clean_fem_preg', 'clean_fem_resp', 'defaultdict', 'main', 'make_preg_map', 'np', 'pd', 'read_fem_preg', 'read_fem_resp', 'stat1', 'sys', 'validate_pregnum']


In [13]:
caseid = 9357
preg_map = nsfg1.make_preg_map(df_preg)

# Check if caseid is present in the preg_map
if caseid in preg_map:
    indices = preg_map[caseid]
    result = df_preg.outcome[indices].values
    print(result)
else:
    print("No matching entries for caseid", caseid)


No matching entries for caseid 9357


In [15]:
if 'finalwgt' in df_preg.columns:
    print("The 'finalwgt' column is present in the DataFrame.")
else:
    print("The 'finalwgt' column is not present in the DataFrame.")



The 'finalwgt' column is not present in the DataFrame.
